setuptools/dist.py

```diff
@@ -195,8 +195,10 @@ def check_packages(dist, attr, value):
 
 
 if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
     # Work around a mypy issue where type[T] can't be used as a base: https://github.com/python/mypy/issues/10962
-    _Distribution = distutils.core.Distribution
+    _Distribution: TypeAlias = distutils.core.Distribution
```

Comment:
I think this one we can do:

```suggestion
    from distutils.core import Distribution as _Distribution
```

and remove the import for `TypeAlias`.

(`errors.py` should be fine using the `TypeAlias` though - I am indifferent on the approach for that file).

Reply:
Since you already approved and these are changed in https://github.com/pypa/setuptools/pull/4504 , I'll keep it as-is for this PR